src/app/forms/validators/custom-async-validators.ts

```diff
@@ -114,7 +117,62 @@ export class CustomAsyncValidators {
 
           return null;
         })
+   
+        );
+  }
+
+  static hideIfEqualsWithCondition(store: Store<State>, sibling: string, value: string, conditions: Condition | Condition[]): AsyncValidatorFn {
+    return (control: AbstractControl): Observable<ValidationErrors | null> =>
+      store.pipe(
+        take(1),
+        select(testResultInEdit),
+        map(testResult => {
+
+          if(testResult){
+            const conditionsPassed = CustomAsyncValidators.checkConditions(testResult, conditions);
+
+            if (control?.parent) {
+              const siblingControl = control.parent.get(sibling) as CustomFormControl;
+              siblingControl.meta.hide = Array.isArray(value) ? (conditionsPassed && value.includes(control.value)) : (conditionsPassed && control.value === value);
+            }
+          }
+
+          return null;
+        })
       );
   }
- 
+
+  static checkConditions(testResult: TestResultModel, conditions: Condition | Condition[] ){
+    if(Array.isArray(conditions)){
+      let failed: boolean = false;
+
+      conditions.forEach((condition) => {
+        if(!CustomAsyncValidators.checkCondition(testResult, condition)) failed = true;
+      });
+
+      return !failed;
+    }else{
+      return CustomAsyncValidators.checkCondition(testResult, conditions);
+    }
+  }
+
+  static checkCondition(testResult: TestResultModel, condition: Condition ){
+
+    const {field, operator, value} = condition
+
+    let fieldValue: any;
+
+    if(testResult?.testTypes[0].hasOwnProperty(field)) {
+      fieldValue = (testResult?.testTypes[0] as any)[field];
+    }else if(testResult?.hasOwnProperty(field)) {
+      fieldValue = (testResult as any)[field];
+    }
+
+    if(operator === operatorEnum.Equals){
+        return Array.isArray(value) ? value.includes(fieldValue) : fieldValue === value
+      }else if(operator === operatorEnum.NotEquals){
+        return Array.isArray(value) ? !value.includes(fieldValue) : fieldValue !== value
+      }
+      return false;
```

Comment:
This can be shortened.
```suggestion
    const {field, operator, value} = condition

    const fieldValue = testResult.testTypes[0].hasOwnProperty(field)
      ? (testResult?.testTypes[0] as any)[field]
      : (testResult as any)[field];

    const isTrue = Array.isArray(value) ? value.includes(fieldValue) : fieldValue === value;

    return operator === operatorEnum.Equals ? isTrue : !isTrue;
```
You don't need to worry about whether `testResult?.hasOwnProperty(field)` fails or not, because if it does, `fieldValue` will be undefined anyway and fail the lower test.